src/eva/data/ioda_obs_space.py

```diff
@@ -139,6 +144,10 @@ def execute(self, data_collections):
                         rename_dict[group_var] = group_name + '::' + group_var
                     ds = ds.rename(rename_dict)
 
+                    # Reset channel numbers from header
+                    if 'nchans' in ds.keys()
```

Comment:
I'm not sure this will work, have you tested it? Because `ds.keys()` is for the group which may not contain the header information? But I'm not sure.